clang/test/SemaSYCL/mark-kernel-name.cpp

```diff
@@ -1,9 +1,13 @@
 // RUN: %clang_cc1 %s -std=c++17 -triple x86_64-linux-gnu -Wno-sycl-2020-compat -fsycl-is-device -verify -fsyntax-only -Wno-unused
 
-template <typename KernelName, typename KernelType>
-[[clang::sycl_kernel]] void kernel_single_task(const KernelType &kernelFunc) { // #kernelSingleTask
-  kernelFunc();
-}
+#include "Inputs/sycl.hpp"
+
+// Test to validate that __builtin_sycl_mark_kernel_name properly updates the
+// constexpr checking for __builtin_sycl_unique_stable_name. We need to make
+// sure that the KernelInfo change in the library both still stays broken, and
+// is then 'fixed', so the definitions below help ensure that is the case.
+// We also validate that this works in the event that we have a wrapper that
+// first calls for the KernelInfo type, then instantiates a kernel.
 
 template <typename KN>
```

Comment:
Aren't these definitions present in mock headers?

Reply:
No, they aren't.  And they shouldn't be, because they are the old/new behavior of the library.  I wouldn't want a future change in the mock library to lose the validation of this behavior.

Reply:
Ok. Thanks for clarifying!